independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/CollectionHelpers.java

```diff
@@ -13,16 +13,20 @@
         //do not instantiate
     }
 
+    @SuppressWarnings("unchecked")
     static <T> Set<T> toImmutableSmallSet(Set<T> set) {
         if (set == null)
             return null;
         switch (set.size()) {
             case 0:
-                return Collections.emptySet();
+                return Set.of();
             case 1:
-                return Collections.singleton(set.iterator().next());
+                return Set.of(set.iterator().next());
+            case 2:
+                Iterator<T> it = set.iterator();
+                return Set.of(it.next(), it.next());
```

Comment:
hum, `Set.of( )` produces an object of 24 bytes, while `Collections.singleton` 16 ,,. previous was better?

Reply:
So I suppose you mean the `java.util.ImmutableCollections.Set12` which is used for both a singleton and a doubleton.

1. Previously, we did not use a specialized set for a doubleton, which is a pity because in CDI doubletons are quite common.
2. I don't think it's reasonable to sacrifice the consistency for 8 bytes...